### PR TITLE
Closes #106 Updated README.md with correct instructions for adding new SDTM datasets

### LINKED ISSUE #106: Update readme instructions on updating sdtm data
@bms63 @rossfarrugia I followed instructions in README file, but I can't generate the `.man` files, it ERRORs each time I run `devtools::document()`. The instructions are out of date, as the programs are in `dev` folder not `inst/dat_scripts`, so maybe there have been other changes?? Any guidance appreciated, thanks.

_Originally posted by @millerg23 in https://github.com/pharmaverse/admiral.test/issues/104#issuecomment-1408790555_

Definition of Done

- [ ] Instructions for updating sdtm datasets are updated and clear.
- [ ] A list of recommended packages to install `metatools`, `labelled` others for development work
- [ ] A brief rationale on why we don't include these packages in DESCRIPTION file or renv.lock

### PR DESCRIPTION
**Pull request by fshanlee**
and updating existing SDTM datasets.

Thank you for your Pull Request! 

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. 

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
Can you add a NOTE:  around developers will need to install packages not currently availabe in the lock file, e.g. metatools.  Can you look through the scripts and see if there are others.

We are trying to keep this package lean and mean and don't want to put them in the suggests or imports of the DESCRIPTION

**Comment by fshanlee** [comment]
> Can you add a NOTE: around developers will need to install packages not currently availabe in the lock file, e.g. metatools. Can you look through the scripts and see if there are others.

Just to clarify: If a developer needs to use a package in order to create/update their _SDTM_ data, and if that package has not been specified in the `renv` lock file, then they will need to install that package. However, if their code does not require a package, then there is no requirement to install it--i.e. they don't need to install {metatools} unless their program uses {metatools} etc. So the _Read Me_ instructions should **not** specify a list of packages that developers will need to install; instead, we should instruct developers to install any packages that they require in addition to those specified in the lock file--is that correct?

**Comment by bms63** [comment]
> > Can you add a NOTE: around developers will need to install packages not currently availabe in the lock file, e.g. metatools. Can you look through the scripts and see if there are others.
> 
> Just to clarify: If a developer needs to use a package in order to create/update their _SDTM_ data, and if that package has not been specified in the `renv` lock file, then they will need to install that package. However, if their code does not require a package, then there is no requirement to install it--i.e. they don't need to install {metatools} unless their program uses {metatools} etc. So the _Read Me_ instructions should **not** specify a list of packages that developers will need to install; instead, we should instruct developers to install any packages that they require in addition to those specified in the lock file--is that correct?

I might be cautious around allowing developers to install any package they desire to develop the test data.  Currently, the only packages not in the lock file are  `metatools` and `ggplot2`.  I haven't done a full review of the code.  Could you please look through?  There was labelled but I am hoping to replace that with metatools.  So providing the list is still helpful and a NOTE on discussing with core-team around using other packages before implementing in their code.

I just don't want to deal with a package that is kicked off CRAN and not accessible down the road.

**Comment by fshanlee** [comment]
@pharmaverse/admiral , in {admiral.test}, many of the programs in the `dev/` folder load {admiral} and {admiral.test}, i.e.:

```
library(admiral)
library(admiral.test)
```

**Is it correct to load {admiral.test} at the start of a program that generates data that will become part of {admiral.test}?** If there are dependencies, for example, the output from `tr.R` is the input for `tu.R`, and if both programs are updated before the package is built, then is there a risk that `tu.R` will read an out-of-date version of the output from `tr.R`, i.e. the `admiral_tr` dataset that was created when the package was last built, rather than the `admiral_tr` dataset that is created by running the updated `tr.R`?

**Comment by fshanlee** [comment]
Hi @bms63 , `pc.R` loads {plyr} in addition to {dplyr}. The {plyr} package is not specified in `renv.lock`, but I have not yet mentioned it in README.md. Do you know anything about {plyr}? Should we mention it in addition to {metatools} and {ggplot2}?

`library(plyr)`

**Comment by bms63** [comment]
> Hi @bms63 , `pc.R` loads {plyr} in addition to {dplyr}. The {plyr} package is not specified in `renv.lock`, but I have not yet mentioned it in README.md. Do you know anything about {plyr}? Should we mention it in addition to {metatools} and {ggplot2}?
> 
> `library(plyr)`

I don't think we need `plyr`.  I think that was an early version of `dplyr`

File: inst/WORDLIST [hunk 1]
```diff
@@ -28,6 +29,7 @@ SUPPTR
 SV
 anonymized
 funder
+ggplot
```
Comment (bms63) [inline inst/WORDLIST:32]: I think we want ggplot2 not to throw an error for the Spelling check so we should leave it in.

**Comment by bms63** [comment]
@millerg23 @bundfussr @rossfarrugia Shan is updating the instructions.   Can we get you to take a look at these as well as you three have been the most active on this package.

**Comment by rossfarrugia** [comment]
> @millerg23 @bundfussr @rossfarrugia Shan is updating the instructions. Can we get you to take a look at these as well as you three have been the most active on this package.

I'm out of touch on this one sorry Ben - would take me some time to dig back into the details so hopefully one of the others could take a look

File: README.md [hunk 2]
```diff
@@ -21,18 +21,27 @@ remotes::install_github("pharmaverse/admiral.test", ref = "devel")
...
+-   Reflect this update, by specifying `admiral_<name>` in `R/data.R`.
+-   Run `devtools::document()` in order to update `NAMESPACE` and update the `.Rd` files in `man/`.
 
 ## Updating Existing SDTM Datasets
 
-- Rename the source dataset as `raw_<name>` (e.g. `raw_ds`) and reflect this change in `R/data.R` by including both `raw_<name>` and `admiral_<name>` here.
-- Create a program at `inst/data_scripts` named `update_<name>.R` to read in `raw_<name>`, make the updates, and output as `admiral_<name>`.
-- Add the output dataset to `data` folder
-- Run `devtools::document()` and include the updated `NAMESPACE` and `.Rd` files in `man/`
+-   Rename the source dataset as `raw_<name>.rda`, where `<name>` is the SDTM domain name, (e.g. `raw_ds.rda`), and copy it to the `data/` folder.
```
Comment (bundfussr) [inline README.md:43]: Just renaming the file does not work. If you rename `admiral_dm.rda` to `raw_dm.rda` and call `data("raw_dm")` you still get `admiral_dm`. You have to recreate the `.rda` file with the new dataset name.

File: README.md [hunk 3]
```diff
@@ -21,18 +21,27 @@ remotes::install_github("pharmaverse/admiral.test", ref = "devel")
...
+-   Each of these programs is written as a standalone R script: if any packages need to be loaded for a given program, then call `library()` at the start of the program.
+-   Most of the packages that you are likely to need will already be specified in the `renv.lock` file, so they will already been installed if you have been keeping in sync--you can check this by entering `renv::status()` in the Console. However, you may also wish to install {metatools} and {ggplot2}, which are currently not specified in the `renv.lock` file. If you feel that you need to install any other packages in addition to those just mentioned, then please tag `@pharmaverse/admiral` to discuss with the development core team.
+-   When you have created a program in the `dev/` folder, you need to run it as a standalone R script, in order to generate a test dataset that will become part of the {admiral.test} package, but you do not need to build the package.
+-   The programs in `dev/` are stored within the {admiral.test} GitHub repository, but they are **not** part of the {admiral.test} package--the `dev/` folder is specified in .Rbuildignore.
```
Comment (bundfussr) [inline README.md:30]: I would use backticks for `.Rbuildignore` to be consistent. Then we also do not need to add it to `WORDLIST`.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (fshanlee) [reply README.md]: @bundfussr , I've clarified by adding a bullet point to the general instructions and I've amended this line of text.

File: README.md (hunk 3 above)
Comment (fshanlee) [reply README.md]: @bundfussr , I've added back ticks around `.Rbuildignore` and also around the names of packages, so I was able to remove both "Rbuildignore" and "ggplot" from the WORDLIST file.

File: inst/WORDLIST (hunk 1 above)
Comment (fshanlee) [reply inst/WORDLIST]: @bms63 , specifying "ggplot2" in WORDLIST did not seem to work, but I've now added back ticks around names of packages, so it's no longer necessary to specify "ggplot" or "ggplot2" in the WORDLIST file.

File: README.md [hunk 4]
```diff
@@ -21,18 +21,28 @@ remotes::install_github("pharmaverse/admiral.test", ref = "devel")
...
+-   When you have created a program in the `dev/` folder, you need to run it as a standalone R script, in order to generate a test dataset that will become part of the {admiral.test} package, but you do not need to build the package.
+-   Each dataset is stored as a `.rda` file whose name is consistent with the name of the dataset: for example, the dataset `dm` should be renamed to `raw_dm` before saving it as `raw_dm.rda`; if you save `dm` as `raw_dm.rda` and subsequently load the `.rda` file, then `dm` (not `raw_dm`) will be loaded into the global environment.
+-   The programs in `dev/` are stored within the {admiral.test} GitHub repository, but they are **not** part of the {admiral.test} package--the `dev/` folder is specified in `.Rbuildignore`.
+-   When you run a program that is in the `dev/` folder, you generate a dataset that is written to the `data/` folder, which will become part of the {admiral.test} package.
```
Comment (zdz2101) [inline README.md:32]: Perhaps include a link to the [exported data chapter](https://r-pkgs.org/data.html#sec-data-data) in the r-packages book use as a guide/best practices and if they need further details?

Comment (fshanlee) [reply README.md]: @zdz2101 , thanks for your suggestion. I've now added a link to the "Exported data" chapter.

**Review by bundfussr — APPROVED** [review]

**Comment by bms63** [comment]
Thanks Shan!

**Pull request merged**